contracts/TokensReceiver.sol

```diff
@@ -0,0 +1,39 @@
+pragma solidity ^0.6.0;
+import "@openzeppelin/contracts-upgradeable/proxy/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+import "./cryptoblades.sol";
+import "hardhat/console.sol";
+
+contract TokensReceiver is Initializable, AccessControlUpgradeable {
+    bytes32 public constant GAME_ADMIN = keccak256("GAME_ADMIN");
+
+    CryptoBlades public game;
+
+    modifier restricted() {
+        _restricted();
+        _;
+    }
+
+    function _restricted() internal view {
+        require(hasRole(GAME_ADMIN, msg.sender));
+    }
+
+    function initialize(
+        address gameContract
+    ) public initializer {
+        __AccessControl_init_unchained();
+        _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+
+        game = CryptoBlades(gameContract);
+    }
+
+    function fight(uint256 char, uint256 wep, uint32 target, uint8 fightMultiplier) external payable {
+        uint256 offset = game.fight(char, wep, target, fightMultiplier);
```

Comment:
Was this a contract size issue that it couldn't be implemented in game contract?
Having an external call on fight will up the transaction costs and make it harder to profit.

Reply:
Yes, previous contract size was around 23.3 kb, marking the function as payable increased it to around 24.8, which i wasn't able to trim down.